### PR TITLE
refactor camera to use view model

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Migrate a couple existing classes to the UIViewModel.

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraViewModel.kt [hunk 1]
```diff
@@ -1,33 +1,77 @@
...
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.component.location.LocationBehavior
+import com.mapbox.navigation.dropin.lifecycle.UIViewModel
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
-// This might be temporary just to test some camera stuff
-internal class CameraViewModel : ViewModel() {
+sealed class CameraAction {
+    data class UpdateLocation(val location: Location) : CameraAction()
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraViewModel.kt:14]: nit: Seems like this should go into the LocationViewModel 🤔

**Review by kmadsen — APPROVED** [review]
Awesome work to also remove RecenterButtonBehavior

Nice example of sharing data between components with a view model

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraViewModel.kt (hunk 1 above)
Comment (abhishek1508) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraViewModel.kt:14]: Yeah that's why I think this should be merged first and then you could take a look at it in https://github.com/mapbox/mapbox-navigation-android/pull/5533

Comment (abhishek1508) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraViewModel.kt:14]: I know the solution to this. We just need to remove this action and `onAttached` completely from `CameraViewModel`. Make the `CamreraComponent` also dependent on `LocationViewModel`

@kmadsen

**Pull request merged**